lib/smack/SplitStructLoadStore.cpp

```diff
@@ -0,0 +1,137 @@
+#include "smack/SplitStructLoadStore.h"
+#include "llvm/IR/InstIterator.h"
+
+namespace smack {
+
+using namespace llvm;
+
+std::vector<Value*> getFirsts(std::vector<std::pair<Value*, unsigned>> lst) {
```

Comment:
I think this function is fine because C++ does not offer simple built-in functions to do mapping from one list to another. The methods discussed in the link also require at least a function to be written.